### DISCUSSION TITLE
Ability to create account with email address

### DISCUSSION THREAD

**Discussion by a1270** [civitai/civitai#76] [Ideas]
This is made even more stupid by the fact you are required to provide an email to complete the registration. An email address which cannot be changed/removed and isn't displayed anywhere on the account settings.

Stating what you are doing with the collected email addresses in the TOS is a necessity at the least. What exactly are the email addresses being used for? Will they be sold to 3rd parties?

tl;dr
you are already collecting email addresses and asking for user names. There is little reason we need to be forced to compromise privacy/security by linking to a 3rd party site.

**Comment by JustMaier** [comment]
The reason we didn't include email/password accounts is because we didn't want to deal with sending emails for things like password reset. By only using other authentication providers we can avoid the support hell that is "I can't get into my account," instead that's all shifted to the services you can login with.

As for TOS and emails, we didn't include anything because the TOS we based ours on (Hugging Face and Imgur) didn't include anything about it either. We don't currently have any plans to use email addresses, but they are used by the authentication package we use ([NextAuth](https://next-auth.js.org/)) to ensure accounts are unique.

I don't like email. I don't like getting it and I don't want to send it. I'll dig into our authentication system to see if there is a way for us to disable the need for an email all together. Would that be a reasonable resolution for you?

**Reply by a1270** [reply]
>I don't like email. I don't like getting it and I don't want to send it. I'll dig into our authentication system to see if there is a way for us to disable the need for an email all together. Would that be a reasonable resolution for you?

That seems reasonable. I am rather anti-3rd party auth as i had a ton of issues when OAuth was supposed to make logins a thing of the past. All it meant was when a provider went down it was like pulling teeth to get your accounts working again. Transfers were always promised but that didn't matter when the server was just shut off. But I do see the appeal of of just having another site handle it for you. Logins always cause issues. 

thank you.

**Comment by LumiWasTaken** [comment]
I fully support the idea to intruduce an alternative method. 

So far the whole civitai as a centralized place is cool but on the other side i avoid it like the plague since having everything connected with eachother is not cool. Maybe not because of civitai using it maliciously but rather reddit now having a connection between those two accounts.

The most simple thing would be to only have, usename and password and email for recovery only as optional. 
If people dont want to add an email, then dont let them.

Alternatively make it social media 0Auth login and only username and password but putting a large disclaimer there that if you choose a username and password without a 0Auth method you risk losing your password without an option to reset it.


Dont make your personal preferences a user issue. "i dont like emails" is not a valid way to talk your way out of a world that is relying on email as its the most robust atm.

**Reply by JustMaier** [reply]
Totally fair, I shouldn't put my distaste for email on everyone else 😛

Good idea regarding the username/password recovery. I'll discuss it with the team

**Reply by ghost** [reply]
Nice, I was just about to create "Change password functionality" ticket, but you seem to have already discussed this. Either way, it feels weird to have email/password to authorize and also have no way to reset active password. Hope the issue is still on!  :pray:

**Comment by JustMaier** [comment]
This is live! You can now login with an email address 😊

**Reply by Vigilence** [reply]
I tried to create an account, but no activation email was sent to me.  Can't find anything in the spam folder, waited a 2 hours.

**Reply by GregHilston** [reply]
Same thing is happening to me as well

**Reply by JustMaier** [reply]
We'll be upgrading our email service provider today. That should help quite a bit with deliverability. I'll reply here as soon as that's done

**Reply by JustMaier** [reply]
Ok, we've just changed email providers. Can you give it another try?

**Comment by ostap667inbox** [comment]
Sorry, but the authorization system on the site can hardly be called adequate.
1. There is no password recovery system.
2. I forgot the email I used to register. I enter one of my emails at random and a new account is immediately created for me. The modal window requires me to make up a login and agree to the terms and conditions. I can't even close it to unlog out and try to sign in again! What an idiotic thing to do!
3. Going into my account settings I also can't see exactly how I'm logged in, from which email address. What the hell?!

**Reply by ostap667inbox** [reply]
Here, I know my nickname, but I don't remember my email. How do I regain access? All of them? I lost access to my models?!

**Comment by jhlchu** [comment]
Also, every time you guys 503 it de-auths the OAuth token and logs me out of every civitai page in tabs. And you guys been getting 503'd alot recently.

**Reply by WASasquatch** [reply]
Should be able to just refresh. Never been actually logged out of CivitAI in months... That is, I've never logged in but when I registered.

**Comment by space-nuko** [comment]
I dunno if passwordless login is a good tradeoff, issue for me is the UX is disruptive

1. I see a model I want to download, but Civit gates it behind their login because it's marked "sensitive"
2. Okay so I can log in, but I can only do so from email, so I have to pull myself away from the login screen all the way to my email which I'm usually not logged into
3. Then I have to dig through my inbox to see if the login email was delivered
4. The most annoying thing, the login link does *not* redirect to the model I wanted to see, it goes to the home page! I have to go back to the original tab instead, the tab opened by the login link is essentially useless
5. Then refreshing the tab that blocked me from seeing the model to begin with redirects to the home screen after I've logged in too! I have to be careful to go back in my history to land on the model page
6. Finally I can look at the model.

Whereas if you let us add passwords, the process is

1. See gated model
2. Go to login, autofill username and password, submit
3. Be redirected to model page

And yes, the API technically allows you to bypass the gating mechanism, but I don't think most users are going to bother to work around it

If as you say you "didn't want to deal with sending emails for things like password reset", "don't like email", "don't like getting it" and "don't want to send it," then why do every time you make users to have to login to browse a model you send them an email? Seems like it adds more friction to the process than is necessary. I'd rather deal with needing a password reset once or twice than having to put up with going through an email flow *every single time* I need to login

And personally I do not want to have to use a social account to log in, it's a privacy concern from my standpoint

**Comment by crkinard** [comment]
Love how there is now an e-mail login (hate sites that force you to link through social media which I avoid). but the e-mail never shows up. Or at least waiting 15 minuets and it still has not arrived.

**Comment by Shahius** [comment]
So...
It's May 2024 and you still don't have password system?
Just registered with my email and was looking for Change Password option in my account.
Nothing. Nada. Zero.
How come this site doesn't have simple basic functionality?
It feels unsafe to me (doesn't mean it is REALLY unsafe, but this feeling is very unpleasant, like you don't have control).  Just gross.
This is why I am going to look for an alternative, so far...

**Comment by WereCatf** [comment]
It's quite frankly entirely irresponsible that you allow people to add payment methods and all, but no way of changing email address or password or anything. If someone were to get ahold of the account, there'd be basically nothing you could do to thwart them aside from deleting your account, it seems.

**Comment by shodanx2** [comment]
Login + password would be nice, I wipe all cookies on browser close and on a regular basis. 
Firefox has no provision for an exception to keep the session cookies only and only for certain websites, so it gets deleted all the time
I would like to be able to login out of my password manager automatically rather than doing the email boogaloo every time.
There are other options, such as one time password where I keep the seed in my password manager.
But login and password, that's tested tried and true

Someone like scaleway does it in a way I like. 
By default they send you an email link
But there's another option to login with password if you've set it up 

Of course there's also FIDO keys, which I am a bit weary of because they identify the person rather than the account 

What I would love is login with a X.509 smartcard, they're cheap, they're reliable (it's the tech in your credit card) and they're impersonal (it's the person with the card login in, not the identified owner of the fido key )

I explored the nuance between smartcard and fido keys here, if you are curious
https://chatgpt.com/share/67031e22-04d8-8005-a865-c3a44350079b

**Reply by 1Dragoon** [reply]
"X.509 smartcard" is kind of vague. It sounds like you're looking for a PIV card, but unless you have a PCSC card reader (I do, ISO7816 and ISO14443 because I work with these professionally) then you're asking for a major inconvenience. If all you want is PIV, you might want just a yubikey as it literally emulates a PCSC smartcard reader, and PIV is one of the applications it supports, in addition to FIDO2, which most people use it for. (Ask me how I got a bug bounty from Microsoft because they were incorrectly parsing the command you send to the card reader to select FIDO.)

Chatgippity, as basically all chatbots do, is misleading you quite a bit here. FIDO does NOT identify the owner of the key, rather when you create a new credential, it generates a unique private key and accompanying public key that the authenticator (i.e. the remote site) uses for you to prove you're whoever you say you are, which can be a totally anonymous identity. Github has no idea who I really am, yet I authenticate with a FIDO2 key. All github knows about me is an email address I gave them. That's it. The public key they authenticate me with is very much ephemeral, i.e. I can wipe it off of my FIDO2 key, and if I use it again on their site, they'd have no idea that it was the same key I used previously. Moreover, so long as your web browser is being truthful to your key about what domain it is authenticating to, another site from another domain can't have even the slightest idea that you're using this same physical key with any other site.

PIV actually makes it easier to follow somebody online in a way that FIDO keys cannot. PIV only has a limited number of slots, depending on the implementation likely only three or so, meaning if you wanted more anonymous identities you'd need a lot more cards. PIV also is built around X.509 certificates, and by design, the X.509 certificate DOES contain information that identifies you, as that is literally the point of PIV: Being able to authenticate you and know who you are without needing to maintain a database of users -- so long as you trust the issuing certificate, you can be confident that this person is who they say they are, because the issuer vouched for them. But that's not just PIV, that's how PKI is meant to work, thus X.509 certificates are used. Any X.509 system will work exactly this way. If that's not what you want, then you DO NOT want X.509 authentication.

However, for FIDO2, the authenticator only has whatever you voluntarily give them, plus the public key that matches your private key. FIDO2 also has a minimum (per the spec) of 25 credential stores, with 2.1 raising that to 50. Did I also mention, by the way, that FIDO2 is highly resistant to phishing? If somebody tricks you into logging into a fake login page, the wrong authentication hash will be generated, and whoever tries to use it will fail if they try to present it to the real login page to spoof your identity.

Though if you're concerned about losing your physical FIDO2 key, or even better, if you want "virtual" keys that are easily disposable, then use a password manager that supports passkeys. My setup works like this: I have a yubikey on my keyring, a cryptnox FIDO2 card in my wallet, and a thetis key at home connected to my desktop. I register all three of those with sites I really want to keep secure, like my protonmail account, my password manager, and some others. For most everyday purposes, including e.g. Amazon, I use passkeys. I even have two different passkeys for my two different Amazon accounts, and all I do is select the one I want to use, and from wherever, including my phone, my laptop, etc. Meanwhile, if I lose a key, I can use the other two to replace it with a new one, and there's only a handful of sites I would need to do this with. The way I use these keys, by the way, is they require a PIN to be able to authenticate, so somebody stealing it would need to know my PIN before I remove that key from my accounts, and they only get so many PIN retries before it locks them out.

**Reply by shodanx2** [reply]
Hi, 
Thanks for the in depth reply, 
I don't remember what civitai is but 
My problem with FIDO keys is that
even though you say 
"the authenticator only has whatever you voluntarily give them"
from my user perspective the FIDO key has a button and the only consent signal I have is to press the button
I don't know what the software is doing and if it's reading more than what I want from the FIDO key
What I like about smartcard is that, they only contain my certificate, and when I swipe them on the reader, I know it can't read anything else about any other identities. 

I like that each card is a physically separate object for each identity. I don't have to trust the FIDO system to prevent cross account correlations when each card is a standalone physical object.

They're also a lot cheaper than fido keys and being NFC, I'm not tempted to leave them plugged in all the time.

As a user, the separation is feels like a  airgap between identities.

As for fishing, can't PIV cards store both client side and server side certificates ? So both sides have to authenticate to the other and that should make phishing the card impossible.

**Reply by 1Dragoon** [reply]
`from my user perspective the FIDO key has a button and the only consent signal I have is to press the button
I don't know what the software is doing and if it's reading more than what I want from the FIDO key
What I like about smartcard is that, they only contain my certificate, and when I swipe them on the reader, I know it can't read anything else about any other identities.`

FIDO is a smartcard. I have in my wallet a literal FIDO2 smartcard. USB is another supported transport, as is bluetooth. The button is just for presence detection.

A bare card reader can read anything that the application defines. At my work, I wrote code that literally scans every slot on a PIV card to allow the user to select which certificate they want to authenticate with.

Webauthn, which is a protocol your web browser uses to allow the remote side to communicate with your FIDO2 card at all, doesn't even permit talking to the card directly. It literally cannot read anything other than the public key that webauthn exposes to it.

`They're also a lot cheaper than fido keys and being NFC, I'm not tempted to leave them plugged in all the time.`

You've got the transport confused with the application.

`As a user, the separation is feels like a airgap between identities.`

Webauthn offers a much stronger separation than bare smartcard communication does. I've gone over the specification documents for both numerous times. The method you're favoring has far fewer privacy protections than FIDO2 does, if that is what you're looking for. Also, if you're hoping to authenticate to a website like this using PIV, or any website at all really, the PKI infrastructure does not exist.

`As for fishing, can't PIV cards store both client side and server side certificates ? So both sides have to authenticate to the other and that should make phishing the card impossible.`

I think you're referring to MTLS, which supports PIV as an option. Just to give you an idea, in order to do something like this using a smartcard, you'd need your certificate to be signed by an entity that the website trusts, and then you'd need to load your certificate to your card, which to my knowledge there aren't any free or open source tools to do this, although you could (and I have) write your own. Unlike normal TLS, there is no client side public key infrastructure, only private ones. This is just asking for web trackers to be able to uniquely identify you no matter where you go, so it's terrible from a privacy perspective. PIV isn't designed for privacy, in most respects it's designed for exactly the opposite. PIV has a totally different intended use case than FIDO2.

In theory you could build an MTLS setup that is phish resistant as it is pretty ironclad in its protection against MitM attacks, but whoever builds the website would need to be careful how they implement that. It isn't like webauthn where that protection is already built into the protocol specification, and the only party who can screw it up would be whoever wrote your web browser.

**Comment by 1Dragoon** [comment]
Is this something still being worked on? I really don't like using third party identity providers.

**Comment by Penguin-Guru** [comment]
I never get prompted for a password when I try to log in. This account system seems incomplete. Civit seems like a well designed platform for the most part but not being able to log in with a password is a fatal flaw. Alternative mechanisms like FIDO and X.509 would be nice too but they are not alternatives. The platform must support passwords for users who don't want to deal with client certs. Using email tokens as a single auth factor is both insecure, unreliable, and inconvenient.
